examples/source/1.components/amp-web-push/static/sw.js

```diff
@@ -256,6 +163,81 @@ function broadcastReply(command, payload) {
       });
 }
 
+
+/**
+  This section of the ServiceWorker deals with SW events: initializing the DB where tokens will be persisted,
+  sending the request for push notifcations to the endpoint, and listening to push events.
+ */
+
+/**
+  Using skipWaiting() to force the waiting service worker to become the active service worker.
+ */
+self.addEventListener('install', () => self.skipWaiting());
+
+/**
+  Creates the DB to store subscription objects and calls clients.claim(), to allow an active
+  service worker to set itself as the controller for all clients within its scope.
+ */
+self.addEventListener('activate', (event) => {
+  event.waitUntil(
+      idb.open('web-push-db', 1, (upgradeDB) => {
+        upgradeDB.createObjectStore('web-push-subcription', {
+          keyPath: 'id',
+        });
+      }).then(() => self.clients.claim())
+  );
+});
+
+/**
+  Listens to push events, and displays a notification, using the payload text.
+ */
+self.addEventListener('push', (event) => {
+  const options = {
+    body: event.data.text(),
+    icon: 'images/notification-flat.png',
+    vibrate: [100, 50, 100],
+    data: {
+      dateOfArrival: Date.now(),
+      primaryKey: 1,
+    },
+  };
+  event.waitUntil(
+      self.registration.showNotification('Push Notification', options)
+  );
+});
+
+/**
+  Intercepts the /send-push request: retrieves the subscription object from indexedDB, and passes it to an endpoint
+  so WebPush messages can be sent.
+  Any other request goes to the network directly.
+ */
+self.addEventListener('fetch', (event) => {
+  if (event.request.url.includes('send-push')) {
+    event.respondWith(
+        idb.open('web-push-db', 1).then((db) => {
```

Comment:
it'd be nice to use async/await everywhere

Reply:
PTAL